### PR TITLE
Add aliases support for legacy package reference projects

### PR DESCRIPTION
**Pull request by nkolev92**
Fixes https://github.com/dotnet/NuGet.BuildTasks/issues/70. 

Design at: https://github.com/NuGet/Home/blob/dev/designs/PackageReference-Extern-Alias.md

Specifically the details about the change here: https://github.com/NuGet/Home/blob/dev/designs/PackageReference-Extern-Alias.md#solution---technical-details

### CODE REVIEW

File: src/Microsoft.NuGet.Build.Tasks/ResolveNuGetPackageAssets.cs [hunk 1]
```diff
@@ -830,6 +831,11 @@ private IEnumerable<ITaskItem> CreateItems(NuGetPackageObject package, string ke
                 item.SetMetadata(NuGetIsFrameworkReference, "false");
                 item.SetMetadata(NuGetSourceType, NuGetSourceType_Package);
 
+                string aliases = file.Value["aliases"]?.ToString();
+                if (aliases != null)
+                {
+                    item.SetMetadata(Aliases, aliases);
+                }
```
Comment (tmeschter) [inline src/Microsoft.NuGet.Build.Tasks/ResolveNuGetPackageAssets.cs:838]: Does this imply that project.assets.json allows aliases on things other than assembly references? Not that I think it matters; I'm just curious.

File: NuGet.config [hunk 2]
```diff
@@ -6,5 +6,6 @@
     <add key="roslyn-tools" value="https://dotnet.myget.org/F/roslyn-tools/api/v3/index.json" />
     <add key="dotnet-core" value="https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json" />
     <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
+    <add key="dotnet-eng" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json" />
```
Comment (tmeschter) [inline NuGet.config:9]: Why is this needed?

Comment (nkolev92) [reply NuGet.config:9]: This repo is arcade powered now, so it needed some dependencies from there. 
Don't recall 100% what the failure was but I can figure it out if you need me to.

File: src/Microsoft.NuGet.Build.Tasks/ResolveNuGetPackageAssets.cs (hunk 1 above)
Comment (nkolev92) [reply src/Microsoft.NuGet.Build.Tasks/ResolveNuGetPackageAssets.cs:838]: Nope, it's mostly a consequence of how the tasks are implemented and I didn't want to refactor too much around it. 

The Aliases will only be specified in the compile groups.

On the SDK side, I implemented on the compile time items only: https://github.com/dotnet/sdk/pull/11451
